File: auth/auth-interceptor/src/main/java/me/nalab/auth/interceptor/ErrorTemplate.java
package me.nalab.auth.interceptor;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;

@Getter
public class ErrorTemplate {

    @JsonProperty("response_messages")
    private final String message;

    private ErrorTemplate(String message) {
        this.message = message;
    }

    public static ErrorTemplate of(String message) {
        return new ErrorTemplate(message);
    }

}

File: gallery/src/main/kotlin/me/nalab/gallery/controller/ErrorTemplate.kt
package me.nalab.gallery.controller

import com.fasterxml.jackson.annotation.JsonProperty

class ErrorTemplate(
    @JsonProperty("response_messages")
    private val message: String,
) {
    companion object {
        fun of(message: String): ErrorTemplate {
            return ErrorTemplate(message)
        }
    }
}

